### ISSUE TITLE
[Content] : Tag field from user returns different data compared to system Tag field

### ISSUE THREAD

**Issue by jcastro-dotcms** [dotCMS/core#19458]
Reported via Support: [102600](https://dotcms.zendesk.com/agent/tickets/102600)
Reproducible **as of dotCMS 5.2.7** up to latest master.

When using a Lucene query to return data, the value of the `+$contentTypeVarName.tagFieldVarName:{tag-name}` causes the query to return different results compared to using the system Tags field:`+tags:{tag-name}`

**To Reproduce**
1. Create a test Content Type called `Test CT` with two fields:
1.1 Text field called Title, required, searchable.
1.2 Tag field called My Tags, required, searchable.
2. Create these two tags:
2.1 Name: `|africa`
2.2 Name: `|africa|algeria`
3. Create two Contentlets of type `Test CT`, one pointing to the `|africa` tag, and one pointing to `|africa|algeria`
4. In the **Content Search** portlet, filter the contents by **Tag** with this value: `|africa`. However, it returns two both contents instead of just one:
<img width="1130" alt="query-content-search" src="https://user-images.githubusercontent.com/8779585/96281196-3db95800-0f96-11eb-8c71-c60f434cb20f.png">

The **Show Query** modal indicates the Lucene query used to retrieve the data:
```
+contentType:TestCt +(conhost:48190c8c-42c4-46af-8d1a-0cd5db894797 conhost:SYSTEM_HOST) +TestCt.myTags:"\|africa" +languageId:1 +deleted:false +working:true
```
5. If you copy the query above, go to the **Query Tool** portlet and replace `+TestCt.myTags:"\|africa"` with `+tags:"\|africa"`, you will get the expected behavior now:
<img width="1108" alt="query-tool-data" src="https://user-images.githubusercontent.com/8779585/96281572-bf10ea80-0f96-11eb-9260-0b28a6f8f09f.png">

**Expected behavior**
Using either tag reference must return the same data, as no wildcards are being passed down to the value in order to retrieve more than one result.

**Additional context**
The currently valid workaround is to use the `+tags:{tag-name}` parameter to query for one specific Tag name, and not to get contents whose Tags start with `{tag-name}`

**Comment by jcastro-dotcms** [comment]
### Workaround # 2:

@wezell suggested the following additional workaround, which works as expected using the same reproducible case explained above:

1. Go to the `Test CT` Content Type.
2. Edit the `My Tags` field.
3. Go to the **Field Variables** tab, and add the following custom mapping:
3.1 Key = `esCustomMapping`
3.2 Value = `{"type": "keyword"}`
4. **VERY IMPORTANT:** Kick off a full reindex for the custom mapping to be added to the ES Index.

**Comment by nollymar** [comment]
PR: https://github.com/dotCMS/core/pull/19628

**Comment by bryanboza** [comment]
Fixed, tested with the provided steps and now I'm unable to reproduce the problem. New UI enhancement reported here #19688

**Comment by jcastro-dotcms** [comment]
The official code changes have been included in the dotCMS 5.3.8.4 LTS release only. The 5.2.8.4 LTS release is several commits behind, which made the code back-port not possible.
